config/jobs/etcd-druid/etcd-druid-api-unit-tests.yaml

```
@@ -0,0 +1,64 @@
+presubmits:
+  gardener/etcd-druid:
+    - name: pull-etcd-druid-api-unit
+      cluster: gardener-prow-build
+      always_run: true
+      optional: true
+      skip_branches:
+        - hotfix-v\d+.\d+ # don't run on release branches for now (add a job per branch later)
+      decorate: true
+      decoration_config:
+        timeout: 40m
+        grace_period: 10m
+      annotations:
+        description: Runs unit tests for etcd-druid API in pull requests
+        fork-per-release: "true"
+      spec:
+        containers:
+          # Run all tests sequentially in one container or as separate prow jobs.
+          # Test will fail when 'check-generate' (which includes a revendor) and 'test' or 'check' run in parallel on the same volume
+          - image: europe-docker.pkg.dev/gardener-project/releases/ci-infra/golang-test:v20250129-fcd6b72-1.23
+            command:
+              - bash
+            args:
+              - c
+              - "cd api && make check check-generate test-unit"
```

Hopefully to keep the `make` invocation pid=1 which is important for signal handling. For sure `bash -c` does not propagate the signal (SIGTERM for example) to the child process.

```suggestion
            command:
              - make
            args:
              - --directory=api
              - check
              - check-generate
              - test-unit
```

---

Nice! thanks for suggesting this change.